amundsen_application/base/examples/example_redash_preview_client.py

```diff
@@ -0,0 +1,118 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+
+from typing import Dict, Optional
+
+from amundsen_application.base.base_redash_preview_client import BaseRedashPreviewClient
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+# Redash natively runs on port 5000, the same port as Amundsen.
+# Make sure to update the running port to match your deployment!
+DEFAULT_URL = 'http://localhost:5010'
+
+
+SOURCE_DB_QUERY_MAP = {
+    'snowflake.ca_covid': 1,
+    'snowflake.demo_data': 37,
+    'redshift.marketing': 245
+}
+
+# This example uses a common, system user, for the API key
+REDASH_USER_API_KEY = os.environ.get('REDASH_USER_API_KEY', '')
+
+
+def _build_db_cluster_key(params: Dict) -> str:
+    _db = params.get('database')
+    _cluster = params.get('cluster')
+
+    db_cluster_key = f'{_db}.{_cluster}'
+    return db_cluster_key
+
+
+class RedashSimplePreviewClient(BaseRedashPreviewClient):
+    def __init__(self,
+                 *,
+                 redash_host: str = DEFAULT_URL,
+                 user_api_key: Optional[str] = REDASH_USER_API_KEY) -> None:
+        super().__init__(redash_host=redash_host, user_api_key=user_api_key)
+
+    def get_redash_query_id(self, params: Dict) -> Optional[int]:
+        """
+        Retrieves the query template that should be executed for the given
+        source / database / schema / table combination.
+
+        Redash Connections are generally unique to the source and database.
+        For example, Snowflake account that has two databases would require two
+        separate connections in Redash. This would require at least one query
+        template per connection.
+
+        The query ID can be found in the URL of the query when using the Redash GUI.
+        """
+        db_cluster_key = _build_db_cluster_key(params)
+        return SOURCE_DB_QUERY_MAP.get(db_cluster_key)
+
+
+class RedashComplexPreviewClient(BaseRedashPreviewClient):
+    def __init__(self,
+                 *,
+                 redash_host: str = DEFAULT_URL,
+                 user_api_key: Optional[str] = REDASH_USER_API_KEY) -> None:
+        super().__init__(redash_host=redash_host, user_api_key=user_api_key)
+        self.default_query_limit = 100
+        self.max_redash_cache_age = 3600  # One Hour
+
+    def _get_query_api_key(self, params: Dict) -> Optional[str]:
+        if params.get('database') in ['redshift']:
+            return os.environ.get('REDSHIFT_USER_API_KEY', '')
+        return None
+
+    def get_redash_query_id(self, params: Dict) -> Optional[int]:
+        db_cluster_key = _build_db_cluster_key(params)
+        return SOURCE_DB_QUERY_MAP.get(db_cluster_key)
+
+    def get_select_fields(self, params: Dict) -> str:
+        """
+        Manually defining the dictionary in this function for readability
+        """
+        field_select_vals = {
```

Comment:
The thought here was to show the full extent for which the queries can be annotated with explicit examples since this is an example class. I can move more of this to the readme / doc strings if you feel the example class should be as close to plug-and-play / barebones as possible.

Reply:
resolved synchronously -- decided to keep this in but with description of why